packages/atlas/src/components/_crt/MarketDrawer/MarketDrawer.tsx

```diff
@@ -0,0 +1,251 @@
+import { useMemo, useRef, useState } from 'react'
+import { flushSync } from 'react-dom'
+import { Controller, useForm } from 'react-hook-form'
+import { CSSTransition, SwitchTransition } from 'react-transition-group'
+
+import { SvgActionPlay, SvgAlertsInformative24 } from '@/assets/icons'
+import { CrtDrawer } from '@/components/CrtDrawer'
+import { FlexBox } from '@/components/FlexBox'
+import { NumberFormat } from '@/components/NumberFormat'
+import { ColumnBox } from '@/components/ProgressWidget/ProgressWidget.styles'
+import { Text } from '@/components/Text'
+import { Tooltip } from '@/components/Tooltip'
+import { TextButton } from '@/components/_buttons/Button'
+import { CrtMarketForm } from '@/components/_crt/MarketDrawer/MarketDrawer.types'
+import { MarketDrawerPreview } from '@/components/_crt/MarketDrawer/MarketDrawerPreview'
+import { Checkbox } from '@/components/_inputs/Checkbox'
+import { TextArea } from '@/components/_inputs/TextArea'
+import { TokenInput } from '@/components/_inputs/TokenInput'
+import { SummaryRow } from '@/components/_overlays/SendTransferDialogs/SendTransferDialogs.styles'
+import { atlasConfig } from '@/config'
+import { useConfirmationModal } from '@/providers/confirmationModal'
+import { useJoystream } from '@/providers/joystream'
+import { transitions } from '@/styles'
+import { Divider } from '@/views/global/NftSaleBottomDrawer/NftForm/AcceptTerms/AcceptTerms.styles'
+
+enum MARKET_STEPS {
+  market,
+  saleSummary,
+}
+const marketStepsNames: string[] = ['Market', 'Sale summary']
+
+export type CrtMarketSaleViewProps = {
+  tokenName: string
+  show: boolean
+  onClose: () => void
+}
+
+const DEFAULT_MIN_PRICE = 100
+
+export const MarketDrawer = ({ show, onClose, tokenName }: CrtMarketSaleViewProps) => {
+  const [activeStep, setActiveStep] = useState(MARKET_STEPS.market)
+  const [isGoingBack, setIsGoingBack] = useState(false)
+  const { tokenPrice } = useJoystream()
+  const nodeRef = useRef<HTMLDivElement>(null)
+
+  const {
+    control,
+    getValues,
+    resetField,
+    formState: { isDirty },
+  } = useForm<CrtMarketForm>({
+    defaultValues: {
+      price: DEFAULT_MIN_PRICE,
+      tnc: atlasConfig.legal.crtTnc,
+      isChecked: true,
+    },
+  })
+  const tokenInUsd = (getValues('price') || 0) * (tokenPrice || 0)
```

Comment:
I won't update on input change, RHF works on refs and getValues is lazy. `watch` should fix it